onyo/lib/commands.py

```diff
@@ -1084,6 +1084,67 @@ def onyo_rm(inventory: Inventory,
     ui.print('Nothing was deleted.')
 
 
+@raise_on_inventory_state
+def onyo_rmdir(inventory: Inventory,
+               dirs: list[Path] | Path,
+               message: str | None = None,
+               auto_message: bool | None = None) -> None:
+    r"""Delete empty directories or convert empty Asset Directories into Asset Files.
+
+    If the directory does not exist, the path is protected, or the asset is
+    already an Asset File, then an error is raised nothing is modified.
+
+    Parameters
+    ----------
+    inventory
+        The Inventory in which to delete directories or convert asset directories.
+    dirs
+        Paths of directories to delete or convert.
+    message
+        Commit message to append to the auto-generated message.
+    auto_message
+        Generate a commit-message subject line.
+        If ``None``, lookup the config value from ``onyo.commit.auto-message``.
+
+    Raises
+    ------
+    ValueError
+        ``dirs`` is empty.
+    """
+
+    if auto_message is None:
+        auto_message = inventory.repo.auto_message
+
+    if not dirs:
+        raise ValueError("At least one directory path must be specified.")
```

Comment:
I do wonder whether that should be a `NoopError`, too.
If one programmatically assembles a list of dirs to remove, then ending up with an empty list or a list of things that lead to `NoopError`s is probably not different in how one wants to deal with that.
So:
```
try:
    onyo_rmdir(mylist)
except NoopError:
    # that's fine
    pass
```

would be convenient. Whereas catching `NoopError` and `ValueError` needs string comparison in addition, b/c there could be a different `ValueError` from deeper down that's actually a problem and not "nothing to do".

Reply:
Not a blocker at all, just something that I think would be nice if we get that sort of thing sorted out better.

Reply:
I agree. I have adjusted this for both `rmdir` and `mkdir`. I have not gone through the other commands.

A complete review of all exceptions is its own task.